libs/table/src/lib/directives/tree-grid/tree-grid.cell.directive.ts

```diff
@@ -0,0 +1,179 @@
+import { Directive, ElementRef, HostBinding, HostListener, Input, Optional } from '@angular/core';
+
+import { NgxHasFocusDirective } from '../has-focus-action';
+import { NgxTreeGridCellTarget, NgxTreeGridRowTarget } from '../../interfaces';
+import { NgxTreeGridRowDirective } from './tree-grid-row.directive';
+import { NgxTreeGridDirective } from './tree-grid.directive';
+
+/**
+ * A cell directive to handle navigation according to the WCAG treegrid pattern
+ *
+ * See https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/
+ */
+@Directive({
+	selector: '[ngxTreeGridCell]',
+	standalone: true,
+})
+export class NgxTreeGridCellDirective extends NgxHasFocusDirective {
+	/**
+	 * The parent row of the cell
+	 */
+	private row: NgxTreeGridRowDirective;
+
+	/**
+	 * Marks the cell as focusable, but only by setting its focus programmatically, not by the tab key
+	 */
+	@HostBinding('attr.tabIndex') private canFocus: number = -1;
+
+	/**
+	 * Set focus on the previous cell to the left
+	 */
+	@HostListener('keydown.ArrowLeft', ['$event']) public moveLeft(event: Event): void {
```

Comment:
Just out of interest, why is the `ArrowLeft` listener so much more complicated than the others?

(Might be good to also clarify in the comment)

Reply:
I've added a reference to it in the comment on the if statement.

Basically, arrowLeft on a row opens the row. But because of how focus works and how the Hostlistener works in Angular; calling arrowLeft here opened the detail row which would mean that the detail row opens whenever you navigate back on the cells; which you don't want.

This is not the case for the other arrow keys.